be/src/runtime/data_stream_common.cpp

```diff
@@ -0,0 +1,99 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "runtime/data_stream_common.h"
+
+#include "column/chunk.h"
+#include "common/logging.h"
+
+namespace starrocks {
+
+// channel per [sender_id]
+class PassThroughSenderChannel {
+public:
+    void append_chunk(const vectorized::Chunk* chunk, size_t chunk_size) {
+        auto clone = chunk->clone_unique();
```

Comment:
Must need clone here?

Reply:
Yes. Because for receiver, it can do any modifcation on this chunk. Unless we implement COW, we have to clone for safety.